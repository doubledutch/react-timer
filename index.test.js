import React from 'react'
import SharedTimer from './index'
import renderer from 'react-test-renderer'

const getTime = () => new Date('1983-04-26T00:00:00')
const targetTime = new Date('1983-04-26T01:02:03')

test('SharedTimer without Wrapper renders bare text', () => {
  const component = renderer.create(<SharedTimer getTime={getTime} targetTime={targetTime} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('SharedTimer with past targetTime renders time since', () => {
  const component = renderer.create(<SharedTimer getTime={getTime} targetTime={new Date('1983-04-25T00:00:00')} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('SharedTimer renders m:ss', () => {
  const component = renderer.create(<SharedTimer getTime={getTime} targetTime={new Date('1983-04-26T00:03:45')} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('SharedTimer renders 0:ss', () => {
  const component = renderer.create(<SharedTimer getTime={getTime} targetTime={new Date('1983-04-26T00:00:12')} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('SharedTimer defaults getTime to () => new Date()', () => {
  const component = renderer.create(<SharedTimer targetTime={new Date()} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('SharedTimer defaults targetTime to new Date()', () => {
  const component = renderer.create(<SharedTimer />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

const Timer = props => (
  <SharedTimer
    {...props}
    Wrapper="span"
    wrapperProps={{className: props.className}}
  />
)

test('Timer with span Wrapper renders text in span', () => {
  const component = renderer.create(<Timer getTime={getTime} targetTime={targetTime} className="my-class" />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

const TestComponent = ({children, className}) => (
  <div className={className}>T minus {children}</div>
)

const ComponentWrappedTimer = props => (
  <SharedTimer
    {...props}
    Wrapper={TestComponent}
    wrapperProps={{className: props.className}}
  />
)

test('Timer with component Wrapper renders text in component', () => {
  const component = renderer.create(<ComponentWrappedTimer getTime={getTime} targetTime={targetTime} className="my-class" />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
