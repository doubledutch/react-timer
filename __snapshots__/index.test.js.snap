// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharedTimer defaults getTime to () => new Date() 1`] = `"0:00"`;

exports[`SharedTimer defaults targetTime to new Date() 1`] = `"0:00"`;

exports[`SharedTimer renders 0:ss 1`] = `"0:12"`;

exports[`SharedTimer renders m:ss 1`] = `"3:45"`;

exports[`SharedTimer with past targetTime renders time since 1`] = `"24:00:00"`;

exports[`SharedTimer with past targetTime renders time since 5 min ago 1`] = `"5:00"`;

exports[`SharedTimer without Wrapper renders bare text 1`] = `"1:02:03"`;

exports[`Timer with component Wrapper renders text in component 1`] = `
<div
  className="my-class"
>
  T minus 
  1:02:03
</div>
`;

exports[`Timer with span Wrapper renders text in span 1`] = `
<span
  className="my-class"
>
  1:02:03
</span>
`;
